scripts/lib/CIME/tests/test_cs_status.py

```diff
@@ -0,0 +1,315 @@
+            output = output + self.output_line(test_status.TEST_PASS_STATUS, test_name, phase)
+
+        with self.assertRaises(AssertionError):
+            self.assert_core_phases(output, 'my.test', fails=[])
+
+# ========================================================================
+# Actual test class
+# ========================================================================
+
+class TestCsStatus(CustomAssertions):
+
+    # ------------------------------------------------------------------------
+    # Test helper functions
+    # ------------------------------------------------------------------------
+
+    # An arbitrary phase we can use when we want to work with a non-core phase
+    _NON_CORE_PHASE = test_status.MEMLEAK_PHASE
+
+    # Another arbitrary phase if we need two different non-core phases
+    _NON_CORE_PHASE2 = test_status.BASELINE_PHASE
+
+    def setUp(self):
+        self._testroot = tempfile.mkdtemp()
+        self._output = six.StringIO()
+
+    def tearDown(self):
+        self._output.close()
+        shutil.rmtree(self._testroot, ignore_errors=True)
+
+    def create_test_dir(self, test_dir):
+        """Creates the given test directory under testroot.
+
+        Returns the full path to the created test directory.
+        """
+        fullpath = os.path.join(self._testroot, test_dir)
+        os.makedirs(fullpath)
+        return fullpath
+
+    @staticmethod
+    def create_test_status_core_passes(test_dir_path, test_name):
+        """Creates a TestStatus file in the given path, with PASS status
+        for all core phases"""
+        with test_status.TestStatus(test_dir=test_dir_path,
+                                    test_name=test_name) as ts:
+            for phase in test_status.CORE_PHASES:
+                ts.set_status(phase, test_status.TEST_PASS_STATUS)
+
+    def set_last_core_phase_to_fail(self, test_dir_path, test_name):
+        """Sets the last core phase to FAIL
+
+        Returns the name of this phase"""
+        fail_phase = test_status.CORE_PHASES[-1]
+        self.set_phase_to_status(test_dir_path=test_dir_path,
+                                 test_name=test_name,
+                                 phase=fail_phase,
+                                 status=test_status.TEST_FAIL_STATUS)
+        return fail_phase
+
+    @staticmethod
+    def set_phase_to_status(test_dir_path, test_name, phase, status):
+        """Sets the given phase to the given status for this test"""
+        with test_status.TestStatus(test_dir=test_dir_path,
+                                    test_name=test_name) as ts:
+            ts.set_status(phase, status)
+
+    # ------------------------------------------------------------------------
+    # Begin actual tests
+    # ------------------------------------------------------------------------
+
+    def test_single_test(self):
+        """cs_status for a single test should include some minimal expected output"""
```

Comment:
I see you added a bunch of tests but they don't appear to be getting called from scripts_regression_tests.

Reply:
These tests are picked up via python's unittest discovery mechanism. I have verified that they are run by inserting this change:

```diff
diff --git a/scripts/lib/CIME/tests/test_cs_status.py b/scripts/lib/CIME/tests/test_cs_status.py
index 7d3fba7e9..dea2176b2 100644
--- a/scripts/lib/CIME/tests/test_cs_status.py
+++ b/scripts/lib/CIME/tests/test_cs_status.py
@@ -208,6 +208,7 @@ def test_single_test(self):
         cs_status([os.path.join(test_dir_path, 'TestStatus')],
                   out=self._output)
         self.assert_core_phases(self._output.getvalue(), test_name, fails=[])
+        self.fail()

     def test_two_tests(self):
         """cs_status for two tests (one with a FAIL) should include some minimal expected output"""
```

and verifying that `./scripts_regression_tests.py A_RunUnitTests` leads to a failure.

Reply:
Great. I thought that might be the case.